### DISCUSSION TITLE
Naming of structural streams

### DISCUSSION THREAD

**Discussion by rhodricusack** [automaticanalysis/automaticanalysis#224] [Ideas]
Some projects (such as the CamCAN) want to do multimodal segmentation (i.e., T1 and T2). At the moment, the name for our structural stream is "structural", and there are many modules that use this.

I have a suggestion for extending the stream representation of structurals.

For modules that can sensibly operate on either form of data, or for backwards compatibility, we put
<stream>structural</stream>
For modules that specifically operate on t1 
<stream>t1</stream>
For modules that specifically operate on t2
<stream>t2</stream>

Some modules (e.g., multimodal segmentation) may take both t1 & t2

If a module asks for a "structural" input stream, either because it is a polymath, or because it is old, then it would match either a source stream of "structural" or "t1" or "t2". This could be controlled in aap parameters with something like this:
<options>
   <streammaps>
       <streammap>
           <name>structural</name>
           <matches>structural</matches>
           <matches>t1</matches>
           <matches>t2</matches>
      </streammap>
   </streammaps>
</options>

What do you think?

**Comment by tiborauer** [comment]
We may need another parameter which controls which of t1 and t2 should be used if both present.

**Comment by rhodricusack** [comment]
This could be implemented by changing the mapping above - on a module specific basis in the task list, if necessary.

**Comment by alexvicegrab** [comment]
This sounds like a good solution since, as Rhodri mentioned, it can be implemented at the module and recipe level.

**Comment by jpeelle** [comment]
Sounds good to me! I am glad for the move towards more specific naming of structural images.
